crates/corro-types/src/pubsub.rs

```diff
@@ -384,19 +267,72 @@ struct InnerMatcherHandle {
     cached_statements: HashMap<String, MatcherStmt>,
 }
 
-type MatchCandidates = IndexMap<TableName, IndexSet<Vec<u8>>>;
+pub type MatchCandidates = IndexMap<TableName, IndexSet<(Vec<u8>, i64)>>;
```

Comment:
This is now pretty expensive to lookup against. Previously we could just pass a slice, but now we have to convert to a `Vec<u8>` so we can check `&(vec, cl)`.

Could this be an associated type on the `Handle` trait?

The subs could use `IndexMap<TableName, IndexSet<Vec<u8>>>` and the updates could use `IndexMap<TableName, IndexMap<Vec<u8>, i64>>` and then you wouldn't have to allocate every time you want to check if it's contained.

An associated type looks like:

```rust
trait Handle {
  type CandidatesMatcher;
  // ...
}
```

Then every implementation needs to define that. I think it might work!